pkg/namespacecheck/ownership_test.go

```diff
@@ -0,0 +1,107 @@
+package namespacecheck
+
+import (
+	"encoding/json"
+	"fmt"
+	v1 "github.com/openshift-eng/ci-test-mapping/pkg/api/types/v1"
+	"github.com/openshift-eng/ci-test-mapping/pkg/config"
+	"github.com/openshift-eng/ci-test-mapping/pkg/registry"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestNoOverlap(t *testing.T) {
+	defaultRegistry := registry.NewComponentRegistry()
+
+	componentNameToNamespaces := map[string][]string{}
+	namespacesToComponentNames := map[string][]string{}
+
+	for _, component := range defaultRegistry.Components {
+		// All types currently conform.  If any don't in the future, find a way to handle them.
+		// I recommend not skipping them.  Probably force conformance
```

Comment:
During one of our discussions about this, I was asked by someone (I forget who) to make it so components are free to do whatever they want as long as we can call the Component interface.  

That gives maximum flexibility. A component could pull the mapping from somewhere (i.e. a file).  

I'm fine forcing conformance, though, if we feel it's flexibility we don't need.

Reply:
> I'm fine forcing conformance, though, if we feel it's flexibility we don't need.

I was thinking about adding to the interface requirement a `ListNamespaces`